portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.Login; 
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+    // returns empty string if user is logged in and the login url if user is not logged in  
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        UserService userService = UserServiceFactory.getUserService();
+        String loginUrl=null, logoutUrl=null;
```

Comment:
Nitpick, I think for Java you don't need to initialise a variable to null. They are by default null.

Reply:
done! thanks

Reply:
I see. Thanks for sharing with me. Yes, I searched and found Java to have this requirement for local variables, while instance or class variables are not required. [ref](https://www.java67.com/2016/07/how-to-fix-variable-might-not-have-been-initialized-error-in-java.html#:~:text=This%20error%20occurs%20when%20you,local%20variable%20without%20initializing%20it.&text=If%20compiler%20believes%20that%20a,but%20will%20not%20use%20it.)

I'm not sure how's the convention in Java. in C++, we have this conversion of avoiding multiple variable initialization in oneline. I guess in Java it shall be all right.

Reply:
yea, i initially thought that it should be null too